packages/curve-ui-kit/src/widgets/Disclaimer/Disclaimer.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState, useEffect, useCallback } from 'react'
+import { useNavigate, useLocation } from 'react-router-dom'
+import { t } from '@lingui/macro'
+
+import Stack from '@mui/material/Stack'
+
+import { TabsSwitcher } from 'curve-ui-kit/src/shared/ui/TabsSwitcher'
+import { SizesAndSpaces } from 'curve-ui-kit/src/themes/design/1_sizes_spaces'
+
+import { LastUpdated } from './LastUpdated'
+import { Footer } from './Footer'
+import { TabPanel } from './TabPanel'
+import { Dex } from './Tabs/Dex'
+import { LlamaLend } from './Tabs/LlamaLend'
+import { CrvUsd } from './Tabs/CrvUsd'
+import { SCrvUsd } from './Tabs/SCrvUsd'
+
+const { MaxWidth } = SizesAndSpaces
+
+const tabs = [
+  { id: 'dex', label: t`Dex` },
+  { id: 'lend', label: t`LlamaLend` },
+  { id: 'crvusd', label: t`crvUSD` },
+  { id: 'scrvusd', label: t`Savings crvUSD` },
+] as const
+type TabId = (typeof tabs)[number]['id']
+const DEFAULT_TAB: TabId = 'dex'
+
+const getTabFromUrl = (search: string) => {
+  if (typeof window === 'undefined') return DEFAULT_TAB
+  const params = new URLSearchParams(search)
+  const tabId = params.get('tab')
+  return tabs.find((tab) => tab.id === tabId)?.id ?? DEFAULT_TAB
+}
+
+type Props = {
+  className?: string
+}
+
+export const Disclaimer = ({ className }: Props) => {
+  const navigate = useNavigate()
+  const location = useLocation()
+
+  const [tab, setTab] = useState(getTabFromUrl(location.search))
```

Comment:
it's usually simpler to avoid a state variable and simply take the url as the source-of-truth

Reply:
This is what I did for the table sorting:
https://github.com/curvefi/curve-frontend/blob/c9530c0b13a2e3a63b784f893d5ddffe64f0354d/packages/curve-ui-kit/src/hooks/useSortFromQueryString.ts#L5-L14

Reply:
Cheers, good catch, the state is a leftover from when tabs weren't derived from the URL yet.

* I've removed the state variable and introduced `useTabFromQueryString`.
* I've given up on insisting on a tab property called `id`, as `TabsSwitcher` really wants a `value` property that acts as the tab id. Creating a copy of the tabs for its option parameter with a duplicate property felt nasty.
* However, I had to keep `options={TABS.map((x) => x)}` as I can't pass a `const` array to the `options` parameter, and I need the `as const` for correct type inference for the hook's return values. Instead of type safe unions of ids you'd merely get `string`, losing strict id typing.